### ISSUE TITLE
Remove hard-coded versions

### ISSUE THREAD

**Issue by pgarbe** [projen/projen#1332]
Projen synthesizes sometimes dependencies with a hard-coded version. It's not obvious that the version is managed by projen and makes it hard to work with dependency management tools like dependabot or renovate.

**Comment by Chriscbr** [comment]
I agree this is a bit non-intuitive, although I'm not sure what the right API / experience for this is.

At some level, providing pinned versions is helpful as maintainers of projen since it's easier for us to integration test that new projects work out of the box. This means when you `projen init` a new project, it should be more likely to work out of the box since we can specifically test that the collection of dependencies work together.

On the other hand, it's usually normal / desirable for people to upgrade these dependencies, so there should be an easy transition to this kind of dependency management.

Reopening so we can think about this some more :)

**Comment by RichiCoder1** [comment]
Is there a world where projen bootstrap'd the dependency tree in question, but then let the user manage it from there? With some "migration" mechanism for adding, removing, and enforcing a minumum version for packages?

I know at least for our team, the ability to automate dependency health is incredibly important.
